### PR TITLE
Added 'strict' to the 'assert' module.

### PR DESCRIPTION
**Pull request by DanielRosenwasser**
https://nodejs.org/api/assert.html#assert_strict_mode

CC @rauschma

### CODE REVIEW

File: types/node/node-tests.ts [hunk 1]
```diff
@@ -87,6 +87,8 @@ namespace assert_tests {
         assert.strictEqual(1, 1, "uses === comparator");
 
         assert.throws(() => { throw new Error("a hammer at your face"); }, undefined, "DODGED IT");
+
+        assert.strict.strict.deepEqual([[[1, 2, 3]], 4, 5], [[[1, 2, '3']], 4, 5]);
```
Comment (rauschma) [inline types/node/node-tests.ts]: This test should fail:

* The purpose of assert.strict is that (at least semantically) `assert.strict.equal === assert.strictEqual` (etc.)
* `assert.strict.strict === assert.strict`

You can fix it by replacing deepEqual -> notDeepEqual

Comment (DanielRosenwasser) [reply types/node/node-tests.ts]: The test won't actually get run, it's solely for type checking

**Review by Flarna — APPROVED** [review]

**Comment by Flarna** [comment]
Regarding CI fail see #26063

**Comment by rauschma** [comment]
Can someone please restart this build? Thanks!

**Comment by DanielRosenwasser** [comment]
Finally 🎉

**Comment by rauschma** [comment]
Yay! The typescript-bot has such a friendly name, but is quite ruthless. 😁

**Pull request merged**